oligo_designer_toolsuite/pipelines/_cycle_hcr_probe_designer.py

```diff
@@ -330,402 +411,561 @@ def design_target_probes(
-                    "sequence_encoding_probe_L_rc": str(
+                    + reverse_primer_sequence
+                )
+                new_properties[probe_id]["sequence_dna_template_probe_R"] = (
+                    forward_primer_sequence
+                    + str(
                         Seq(
-                            encoding_probe_database.get_oligo_property_value(
-                                property="sequence_encoding_probe_L",
+                            format_sequence(
+                                database=hybridization_probe_database,
+                                property="sequence_readout_probe_R",
                                 region_id=region_id,
                                 oligo_id=probe_id,
-                                flatten=True,
                             )
                         ).reverse_complement()
-                    ),
-                    "sequence_encoding_probe_R_rc": str(
+                    )
+                    + linker_sequence
+                    + str(
                         Seq(
-                            encoding_probe_database.get_oligo_property_value(
-                                property="sequence_encoding_probe_R",
+                            format_sequence(
+                                database=hybridization_probe_database,
+                                property="sequence_oligo_R",
                                 region_id=region_id,
                                 oligo_id=probe_id,
-                                flatten=True,
                             )
                         ).reverse_complement()
-                    ),
-                }
-        encoding_probe_database.update_oligo_properties(new_probe_properties_primer)
-
-        # Calculate Tm for oligo_pair_L
-        properties = [
-            TmNNProperty(
-                Tm_parameters=self.target_probe_Tm_parameters,
-                Tm_chem_correction_parameters=self.target_probe_Tm_chem_correction_parameters,
-                Tm_salt_correction_parameters=self.target_probe_Tm_salt_correction_parameters,
-            )
-        ]
-        calculator = PropertyCalculator(properties=properties)
-        encoding_probe_database = calculator.apply(
-            oligo_database=encoding_probe_database, sequence_type="oligo_pair_L", n_jobs=self.n_jobs
-        )
-        # Calculate Tm for oligo_pair_R
-        encoding_probe_database = calculator.apply(
-            oligo_database=encoding_probe_database, sequence_type="oligo_pair_R", n_jobs=self.n_jobs
+                    )
+                    + reverse_primer_sequence
+                )
+
+            hybridization_probe_database.update_oligo_properties(new_properties)
+
+        return hybridization_probe_database
+
+    def generate_output(
+        self,
+        probe_database: OligoDatabase,
+        codebook: pd.DataFrame,
+        readout_probe_table: pd.DataFrame,
+        top_n_sets: int = 3,
```

Comment:
I think you tried to remove all default parameter definitions from the pipeline scripts; should the default for top_n_sets then also be removed and only mentioned in the yaml files?